src/main/webapp/script.js

```diff
@@ -652,14 +729,74 @@ function navigateGraph(amount) {
  * Assumes currGraphNum is between 0 and numMutations
  */
 function updateButtons() {
-  if (currGraphNum === 0) {
+  // The use of <= and >= as opposed to === is for safety
+  if (Math.floor(currMutationIndex) <= -1) {
     document.getElementById("prevbutton").disabled = true;
   } else {
     document.getElementById("prevbutton").disabled = false;
   }
-  if (currGraphNum === numMutations) {
+  if (currMutationIndex >= mutationIndexList.length - 1) {
     document.getElementById("nextbutton").disabled = true;
   } else {
     document.getElementById("nextbutton").disabled = false;
   }
+  const numElement = document.getElementById("num-mutation-display");
+  numElement.innerText = `Graph ${currMutationNum + 1}`;
+}
+
+/**
+ * Get the index of the next largest element in an indiceList
+ * @param indicesList a list of indices, assume it's sorted
+ * @param element the element to find a higher value than
+ * @return the index of the first element in indicesList larger than
+ * element or indicesList.length if element is larger than all elements
+ * in indicesList
+ */
+function getIndexOfNextLargerNumber(indicesList, element) {
```

Comment:
I wonder if you could merge `getIndexOfNextLargerNumber` and `getIndexOfPrevSmallerNumber` into the same function. Since you already have an index of the next larger, you could just do a traversal backwards to find the index of the first number smaller than element, right? It should be at most 2 traversals backwards, unless you have multiple elements in the list with the same value (e.g. [5,5,5,5,5] in your test case) but in practice this won't happen for this tool in the real world scenario :)